src/tests/ftest/util/AgentUtils.py

```diff
@@ -111,11 +164,28 @@ def stop_agent(client_list):
 
     client_list -- kill daos_agent on these hosts
     """
-
     if len(sessions) == 0:
         return
 
     # this kills the agent
     for client in client_list:
         cmd = "pkill daos_agent"
-        resp = subprocess.call(["ssh", client, cmd])
\ No newline at end of file
+        resp = subprocess.call(["ssh", client, cmd])
+
```

Comment:
Once you have done the `pkill` on the remote node, the local `ssh` process will die but it's process slot needs to be reaped with a `.wait()` in order to not leave any zombie processes behind.

This is done in the `stopServer()` function like this:
https://github.com/daos-stack/daos/blob/bf7c585cb3727dec197acdc02885eb41dfcaba89/src/tests/ftest/util/ServerUtils.py#L175

The `.kill()` before it:
https://github.com/daos-stack/daos/blob/bf7c585cb3727dec197acdc02885eb41dfcaba89/src/tests/ftest/util/ServerUtils.py#L173-L174

is probably a good measure also.

So you can uses `sessions[client]` here to retrieve the `subprocess` session that you stashed away here:
https://github.com/daos-stack/daos/blob/8a0357d0b856c7c07a9e7eee24807d9a3b031bee/src/tests/ftest/util/AgentUtils.py#L128